### ISSUE TITLE
Allow instantiation of subtheories with opaque operators

### ISSUE THREAD

**Issue by fdupress** [EasyCrypt/easycrypt#154]
The following works.

```
theory T.
type t.

op o : t = witness.
end T.

theory T1.
clone import T.
end T1.

theory T2.
clone T1 as Boo with
  theory T <- T.
end T2.
```

The following doesn't, with the only difference that the operator is made opaque.

```
theory T'.
type t.

op o : t = witness axiomatized by oE.
end T'.

theory T1'.
clone import T'.
end T1'.

theory T2'.
clone T1' as Boo with
  theory T' <- T'.
(* operator `T'.o' body incompatible body *)
end theory T2'.
```

We'd like the second one to go through. (Probably we simply want the compatibility check—in the case of theory instantiation only?—to look under the `axiomatized by`.)

**Comment by chdoc** [comment]
I just ran into this issue as well and lost about one hour before I remembered having seen this. This is really annoying because this bars me from using `axiomatized by` in the new matrix theory where some of the definitions simplify in confusing ways. 

@strub This feels related to #264, can you maybe fix both in one go?

**Comment by fdupress** [comment]
Oh, I was meant to look into this further. My patch worked, but it was brutal and had a big risk of causing performance issues. The proper way of doing it (as discussed in the now closed #157) is to unfold head operators only. I won't be able to look into figuring the code out for this.

The issue in #264 was that the case for program variables in convertibility checks was a "Nope". The fix for that (probably) won't fix this.

**Comment by strub** [comment]
Indeed, #264 is irrelevant for this issue.
